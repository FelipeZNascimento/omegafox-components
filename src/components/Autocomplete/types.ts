export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
