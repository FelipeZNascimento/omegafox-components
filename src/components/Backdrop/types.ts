export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
