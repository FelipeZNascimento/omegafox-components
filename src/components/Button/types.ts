export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
