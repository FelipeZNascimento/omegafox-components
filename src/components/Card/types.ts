export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
