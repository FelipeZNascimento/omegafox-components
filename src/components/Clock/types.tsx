import {
  FOOTBALL_MATCH_STATUS,
  AMERICAN_FOOTBALL_MATCH_STATUS
} from './constants';
import { TBetValues } from '../types';

type TClock = {
  time: number;
};

type FootballKeys = keyof typeof FOOTBALL_MATCH_STATUS;
export type TClockFootball = TClock & {
  status: typeof FOOTBALL_MATCH_STATUS[FootballKeys];
};

type AmericanFootballKeys = keyof typeof AMERICAN_FOOTBALL_MATCH_STATUS;
export type TClockAmericanFootball = TClock & {
  status: typeof AMERICAN_FOOTBALL_MATCH_STATUS[AmericanFootballKeys];
};

export interface IClockProps {
  betValue: TBetValues | null;
  clock: TClockAmericanFootball | TClockFootball;
  isMatchStarted?: boolean;
  isExpandable: boolean;
  isExpanded: boolean;
  isMatchEditable: boolean;
  timestamp: number;
}

export interface IClockMobileProps {
  clock: TClockAmericanFootball | TClockFootball;
  timestamp: number;
}
