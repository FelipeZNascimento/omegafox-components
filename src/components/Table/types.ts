export type TTableColumn = {
  id: number;
  align: 'left' | 'center' | 'right';
  flex: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10;
  renderingFunction: () => JSX.Element;
};

export type TTableRowColumn = {
  id: number;
  renderingFunction: () => JSX.Element;
};

export type TTableRow = TTableRowColumn[];

export interface ITableProps {
  isHeader: boolean;
  isLoading?: boolean;
  columns: TTableColumn[];
  rows: TTableRow[];
}
