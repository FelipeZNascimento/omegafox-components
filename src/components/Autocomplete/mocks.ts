import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
