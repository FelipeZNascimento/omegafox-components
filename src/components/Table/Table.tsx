import React from 'react';
import classNames from 'classnames';

import { ITableProps, TTableRowColumn } from './types';
import styles from './Table.module.scss';
import { Loading } from 'index';
import spinner from '../../img/spinner.png';

export const Table = ({
  columns,
  isHeader,
  isLoading = false,
  rows
}: ITableProps) => {
  const numOfCols = columns.length;

  const renderRow = (rowColumns: TTableRowColumn[], index: number) => {
    return (
      <div key={index} className={styles.tableRow}>
        {rowColumns.map((item: TTableRowColumn, index: number) => {
          //Avoid breaking table by rendering more columns than it should
          if (index >= numOfCols) {
            return null;
          }

          const flexSize = columns[index].flex;
          const itemAlign = columns[index].align;

          const itemClass = classNames(styles[`flexed${flexSize}`], {
            [styles.alignedLeft]: itemAlign === 'left',
            [styles.alignedCenter]: itemAlign === 'center',
            [styles.alignedRight]: itemAlign === 'right'
          });

          return (
            <div key={item.id} className={itemClass}>
              {item.renderingFunction()}
            </div>
          );
        })}
      </div>
    );
  };

  return (
    <>
      {isHeader && renderRow(columns, 0)}
      {isLoading && (
        <div className={styles.loadingContainer}>
          <Loading image={spinner} />
        </div>
      )}
      {!isLoading &&
        rows.map((rowColumns, index) => renderRow(rowColumns, index))}
    </>
  );
};
