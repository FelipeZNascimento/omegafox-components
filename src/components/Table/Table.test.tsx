import React from 'react';
import { render } from '@testing-library/react';
import { Table } from './Table';
import { ITableProps } from './types';
import { tableConfig } from './mocks';

describe('<Table />', () => {
  const renderComponent = ({ isHeader, columns, rows }: ITableProps) =>
    render(<Table isHeader={isHeader} columns={columns} rows={rows} />);

  it('should render', () => {
    const { container } = renderComponent({
      isHeader: tableConfig.isHeader,
      columns: tableConfig.columns,
      rows: tableConfig.rows
    });

    expect(container).toMatchSnapshot();
  });
});
