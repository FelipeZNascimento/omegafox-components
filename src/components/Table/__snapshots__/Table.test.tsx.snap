// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render 1`] = `
<div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        Pos
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Nome
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        Pts.
      </div>
    </div>
  </div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        01.
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Felipe Zanon Felipe Zanon Felipe Zanon
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        999
      </div>
    </div>
  </div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        01.
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Felipe Zanon Felipe Zanon Felipe Zanon
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        999
      </div>
    </div>
  </div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        01.
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Felipe Zanon Felipe Zanon Felipe Zanon
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        999
      </div>
    </div>
  </div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        01.
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Felipe Zanon Felipe Zanon Felipe Zanon
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        999
      </div>
    </div>
  </div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        01.
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Felipe Zanon Felipe Zanon Felipe Zanon
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        999
      </div>
    </div>
  </div>
  <div
    class="tableRow"
  >
    <div
      class="flexed1 alignedLeft"
    >
      <div>
        01.
      </div>
    </div>
    <div
      class="flexed4 alignedLeft"
    >
      <div>
        Felipe Zanon Felipe Zanon Felipe Zanon
      </div>
    </div>
    <div
      class="flexed1 alignedRight"
    >
      <div>
        999
      </div>
    </div>
  </div>
</div>
`;
